### PR TITLE
Support filtering received dynamic manifest

### PR DESCRIPTION
**Pull request by Stebalien**
That way we can, e.g., prevent the manifest server from sending manifests that apply to the mainnet network name, actually finalize tipsets, etc.

See #688.

### CODE REVIEW

File: cmd/f3/observer.go [hunk 1]
```diff
@@ -166,7 +166,10 @@ var observerCmd = cli.Command{
 		if err != nil {
 			return fmt.Errorf("decoding manifestServerID: %w", err)
 		}
-		manifestProvider, err := manifest.NewDynamicManifestProvider(nil, ds, pubSub, manifestServerID)
+		manifestProvider, err := manifest.NewDynamicManifestProvider(
+			pubSub, manifestServerID,
+			manifest.DynamicManifestProviderWithDatastore(ds),
```
Comment (Stebalien) [inline cmd/f3/observer.go:171]: Gets a bit wordy, but makes it easier to add more params later.

**Review by masih — APPROVED** [review]

**Review by Kubuxu — APPROVED** [review]
